src/Bicep.Core/Semantics/NameBindingVisitor.cs

```diff
@@ -74,8 +72,16 @@ public override void VisitModuleDeclarationSyntax(ModuleDeclarationSyntax syntax
             this.Visit(syntax.Name);
             this.Visit(syntax.Path);
             this.Visit(syntax.Assignment);
+            allowedFlags = FunctionFlags.RequiresInlining;
+            this.Visit(syntax.Value);
+            allowedFlags = FunctionFlags.Default;
+        }
+
+        public override void VisitIfConditionSyntax(IfConditionSyntax syntax)
+        {
+            this.Visit(syntax.Keyword);
             allowedFlags = FunctionFlags.Default;
-            this.Visit(syntax.IfCondition);
+            this.Visit(syntax.ConditionExpression);
```

Comment:
This works for now because the `if` syntax is always a direct descendant of resource/module, isn't super obvious, and may be problematic if it's reused elsewhere. Mind adding a note to explain what this is doing and why it's here?

Reply:
Will do.